### DISCUSSION TITLE
C API's vips_resize() "no property named" error

### DISCUSSION THREAD

**Discussion by rafalcode** [libvips/libvips#2517] [Q&A]
## Bug report

**Describe the bug**
libvips C API, the vips_resize() function on JPG files both Debian and Archlinux, using these distributions libvips-dev packages. When running simplest resize program, get ```resize: no property named `d4LV'```. That property name is garbage, as it changes, so it's undefined memory. It also seems to be a gobject error. The documentation for this function also mentions a "double" type for scale and then a "gdouble" type for vscale which are probably the same thing, but seems inconsistent.
Note: on debian, Python binding for resize works perfectly so it's all a little confusing. However I will try doing more tests ... my suspicion is that I am doing something wrong because I imagine this is a heavily used function and so many people not having this problem, just me (my guess).

**To Reproduce**
Compiles fine with appropiate headers and library linking.
Program is almost exact same as one of the examples, except using resize function instead of invert
```
#include <stdio.h>
#include <vips/vips.h>

int main( int argc, char **argv )
{
    VipsImage *in;
    double mean;
    VipsImage *out;

    if( VIPS_INIT( argv[0] ) )
        vips_error_exit( NULL );

    if( argc != 3 )
        vips_error_exit( "usage: %s infile outfile", argv[0] );

    if( !(in = vips_image_new_from_file( argv[1], NULL )) )
        vips_error_exit( NULL );

    printf( "image width = %d\n", vips_image_get_width( in ) );
    printf( "image height = %d\n", vips_image_get_height( in ) );

    if( vips_avg( in, &mean, NULL ) )
        vips_error_exit( NULL );

    printf( "mean pixel value = %g\n", mean );

    // if( vips_invert( in, &out, NULL ) )
    if( vips_resize(in, &out, 1.85) )
        vips_error_exit( NULL );


    if( vips_image_write_to_file( out, argv[2], NULL ) )
        vips_error_exit( NULL );

    g_object_unref( in );
    g_object_unref( out );

    return( 0 );
}
```

**Comment by jcupitt** [comment]
Hi @rafalcode,

You need a `NULL` at the end, so this line:

```C
    if( vips_resize(in, &out, 1.85) )
```

Should be:

```C
    if( vips_resize(in, &out, 1.85, NULL) )
```

If you enable warnings in your compiler it ought to find this error for you. I see:

```
$ gcc -g -Wall rafal.c $(pkg-config vips --cflags --libs)
rafal.c: In function ‘main’:
rafal.c:28:5: warning: not enough variable arguments to fit a sentinel [-Wformat=]
   28 |     if( vips_resize(in, &out, 1.85) )
      |     ^~
```

**Reply by rafalcode** [reply]
Fabulous! many thanks for the extremely speed response. Mea culpa on that .. I was rushing as usual ... didn't have -Wall and didn't wait to the fully read meaning of ellipsis. Many thanks!

**Reply by jcupitt** [reply]
Oh, no problem. The heavy use of varargs in the C API is odd when you first see it.
